controllers/storagecluster/backingstorageclasses.go

```diff
@@ -0,0 +1,91 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type backingStorageClasses struct{}
+
+func constructBackingStorageClass(r *StorageClusterReconciler, bsc *ocsv1.BackingStorageClass) (*storagev1.StorageClass, error) {
+
+	if bsc.Name == "" {
+		return nil, fmt.Errorf("backingStorageClass name is empty")
+	}
+	storageClass := &storagev1.StorageClass{}
+	pvReclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	volumeBindingMode := v1.VolumeBindingWaitForFirstConsumer
+	storageClass.ReclaimPolicy = &pvReclaimPolicy
+	storageClass.AllowVolumeExpansion = &allowVolumeExpansion
+	storageClass.VolumeBindingMode = &volumeBindingMode
+
+	storageClass.ObjectMeta = bsc.ObjectMeta
+	storageClass.Parameters = bsc.Parameters
+	storageClass.Provisioner = bsc.Provisioner
+
+	if bsc.Provisioner == "" {
+		platform, err := r.platform.getPlatform(r.Client)
+		if err != nil {
+			return nil, err
+		}
+		if platform != configv1.AWSPlatformType {
+			return nil, fmt.Errorf("auto detection of provisioner is not supported for %s", platform)
+		}
+		if platform == configv1.AWSPlatformType {
```

Comment:
is this check required?

Reply:
not required anymore as the switch will take care of selecting the right provisioner if not specified.